azext_iot/iothub/providers/helpers/edge_device_config.py

```diff
@@ -0,0 +1,577 @@
+        update-ca-certificates
+fi
+
+systemctl restart docker
+
+# ======================= Copy device certs  =======================================
+cert_dir="/etc/aziot/certificates"
+mkdir -p $cert_dir
+cp "{EDGE_ROOT_CERTIFICATE_FILENAME}" "$cert_dir/{EDGE_ROOT_CERTIFICATE_FILENAME}"
+cp "$device_id.full-chain.cert.pem" "$cert_dir/$device_id.full-chain.cert.pem"
+cp "$device_id.key.pem" "$cert_dir/$device_id.key.pem"
+"""
+EDGE_CONFIG_SCRIPT_HUB_AUTH_CERTS = """
+# ======================= Copy hub auth certs  =======================================
+cert_dir="/etc/aziot/certificates"
+mkdir -p $cert_dir
+cp "$device_id.hub-auth-cert.pem" "$cert_dir/$device_id.hub-auth-cert.pem"
+cp "$device_id.hub-auth-key.pem" "$cert_dir/$device_id.hub-auth-key.pem"
+"""
+EDGE_CONFIG_SCRIPT_APPLY = """
+# ======================= Read User Input =======================================
+iotedge config apply -c /etc/aziot/config.toml
+
+echo "To check the edge runtime status, run 'iotedge system status'. To validate the configuration, run 'sudo iotedge check'"
+"""
+
+EDGE_SUPPORTED_OS_LINK = "https://aka.ms/iotedge-supported-systems"
+EDGE_LINUX_TUTORIAL_LINK = "https://aka.ms/iotedge-provision-linux-device"
+EDGE_WINDOWS_TUTORIAL_LINK = "https://aka.ms/iotedge-provision-windows"
+
+DEVICE_README = f"""
+# Prerequisites
+Each device must have IoT Edge (must be v1.2 or later) installed.
+Pick a [supported OS]({EDGE_SUPPORTED_OS_LINK}) and follow the corresponding tutorial to install Azure IoT Edge:
+    - [Linux on Windows]({EDGE_WINDOWS_TUTORIAL_LINK})
+    - [Linux]({EDGE_LINUX_TUTORIAL_LINK})
+
+# Steps
+
+1. Copy the bundle for each created device (device_id.tgz) onto the device.
+2. Extract the bundle file by running following command:
+
+```Extract
+    tar zxvf ~/<PATH_TO_BUNDLE>/[[device-id]].tgz
+```
+
+3. Run the install script:
+
+```Run
+    sudo bash ./install.sh
+```
+
+4. If hostnames were not provided in the configuration file, the script will prompt for hostnames.
+    - Follow the prompt by entering the device and/or parent hostname (FQDN or IP address).
+    - On the parent device, it will prompt for its own hostname.
+    - On a child device, it may prompt the hostname of both the child and parent devices.
+
+"""
+
+
+def create_edge_device_config(
+    hub_hostname: str,
+    device_id: str,
+    auth_method: DeviceAuthType,
+    device_config: EdgeDeviceConfig,
+    default_edge_agent: str,
+    device_config_path: Optional[str] = None,
+    device_pk: Optional[str] = None,
+    output_path: Optional[str] = None,
+):
+    # load default device TOML object or custom path
+    device_toml = (
+        process_toml_arg(device_config_path)
+        if device_config_path
+        else DEVICE_CONFIG_TOML
+    )
+
+    device_toml[
+        "trust_bundle_cert"
+    ] = f"file:///etc/aziot/certificates/{EDGE_ROOT_CERTIFICATE_FILENAME}"
```

Comment:
is this file path going to always be the same for any system?

Reply:
yes, this is the path the edge agent uses to store certificates and a copy of the device config toml.